src/models/datastore.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright 2015 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+let ModelFactory = require('../helpers/model-factory');
+let Connection = require('./connection');
+
+class DataStore {
+
+  constructor () {
+    this.storeName_ = null;
```

Comment:
What is the scenario where you'd change the origin storeName?

I would have though passing in the only storeName here would make sense.